### ISSUE TITLE
Pytest cache for bls keys

### ISSUE THREAD

**Issue by ChihChengLiang** [ethereum/trinity#773]
#### Step to reproduce

```
branch A: ralexstokes/validator-spec-updates c4019c2e87ee8c85a7fa655b027b2c177ebe2986
branch B: ChihChengLiang/bls-rebase-broken (I'm doing some Chia bls work here)
```

1. checkout A. Run `pytest tests/eth2/core/beacon/test_attestation_helpers.py::test_verify_indexed_attestation_signature[40]`. This suppose to pass
2. checkout B. Run `pytest tests/eth2/core/beacon/test_attestation_helpers.py::test_verify_indexed_attestation_signature[40] --lf`. You can see the test failed
3. checkout A again. Run `pytest tests/eth2/core/beacon/test_attestation_helpers.py::test_verify_indexed_attestation_signature[40]`. This test now failed!

Pytest shows the public keys are cached. Which could be the reason why py_ecc failed.

```
$ pytest --cache-show
======================================================================= test session starts =======================================================================
platform darwin -- Python 3.7.0, pytest-3.6.4, py-1.7.0, pluggy-0.7.1
rootdir: /Users/liangcc/projects/EAPL/eth2.0s/trinity, inifile: pytest.ini
plugins: xdist-1.18.1, trio-0.5.2, timeout-1.3.3, profiling-1.6.0, osxnotify-0.1.7, mock-1.10.4, forked-1.0.2, deadfixtures-2.1.0, cov-2.5.1, asyncio-0.9.0, asyncio-network-simulator-0.1.0a2, hypothesis-3.69.5
cachedir: /Users/liangcc/projects/EAPL/eth2.0s/trinity/.pytest_cache
-------------------------------------------------------------------------- cache values ---------------------------------------------------------------------------
cache/lastfailed contains:
  {'tests/eth2/core/beacon/test_attestation_helpers.py::test_verify_indexed_attestation_signature[40]': True,
   'tests/test_foo.py::test_2': True}
cache/nodeids contains:
  ['tests/eth2/core/beacon/test_attestation_helpers.py::test_verify_indexed_attestation_signature[40]']
eth2/bls/key-cache contains:
  {'pubkeys_by_index': {'0': '95184d510d7f95d0f5a33e0d922639c2660c2bafa7daad1ab87c66711bf66fce1fc9105504902c5b7f784a5ade2c65b9',
                        '1': '868dabc748a064f0323d5ebcb2300f1d5bd2bb972ae09145f72e79495033b1e7d4c622641bf7d14e6a1f679c0ec82f24',
                        '10': 'ae0031515253249cc68e8ff6381c85231781f9ba5c251f8d663d634b461bc6a35ecccd2938704d36cfd7eb7bcf843b82',
                        '11': 'adec245e630cdf805c86f14258ca0e24c8016ec11defb065d70ad9a8c6dee38b26b2499a5cf1931ec65a2640ecd8424b',
                        '12': '956f2f510d8e6acf438600f0bbbf8b6c96e31183abadab8adb864d76dfb209bd3cedad07d188bc53ebcaef76eeb368b1',
                        '13': '9664301dfcb940539b336e6b21c156cda3c8a63847727d37491a892db7f4f2bb9fc9730163254e8bd028b730c8b39309',
                        '14': 'a5b360b364f081836261542b2b89effd6a596ec8b34fd330a80446d64329bcc8c74bce9db9b5c6adcad5f70e3b631bb0',
                        '15': 'a5680dcfa6fe6f7878132df80664232f847e03748b6383a3255c019c45bc9299835562b0b288fa1282c267a94a6daf00',
                        '16': '8494bff563fd1259a04b868f64e0472e2d1f48ff0a02a3142effe17df01f8611146e1218235837868b42963637a4e769',
                        '17': '96fd045abfda74406e9626cb18b8dac331d2ac3b37d8429e16a2f08415e128be210ff16c47c8958043a9812cd78159d9',
                        '18': 'b5f5e6bb054d785fab30756d542e9cd9461b286fc75f09c6368b577f9e3b486d4d1b11fe4c60b3ec838421276b78e55a',
                        '19': '8b3ba6ccb6ce819329a40ed6c804622423dff836f618a364f1b6abfafcf979014cc628c795a65f777ef73de1b60bb9eb',
                        '2': '122b28eb0a7e3727ae9b2b07aedb9b525cf2da080ce4734d9bf253e35a81c33e4e90f6173fd79dade94855ade5b948b4',
                        '20': 'a06ed227d708d67e14dfb5062211b1ee95fd7af61989297e08bc0eb1a9b3c5a9eca256f181b59d09c86746acea06d5e0',
                        '21': '8cfaed1a56f5709324661a8e98718ae1b1a556839d622d473852283887e44aa197c75787d67ec2d1e6ff06a88e4087e4',
                        '22': '90acb49b847d2dc727499c14d667efc7788bfaf12736b80df4ae28c505037e7acf22936964a7a5e75b78d6fdaff196fa',
                        '23': 'a468bd3b3c8e5d281ee78ee82a79fe8a92fa29161a52e9fab6f89d5b504d79bf0a1839262c3ecdef123830a8ac31bb03',
                        '24': 'b4d042a6a4098d9e08bc71ef6be948b6281f0f5a0f827924d367a97dedfd7ad900041dc0dbf45c97908a6b7e7b8de2c1',
                        '25': 'aae77052de5f10b669565187ea3c65ffb21ffb71268b11d1e27ec0e8ded8dee8de2a8316a91d4004d31126bd7e09e0b8',
                        '26': 'b2f3447ee96702a908269eb79948537c86a1d1e425c191a012c5a869ea7aa5d7ef90bea26111d265c8623017b8cb3359',
                        '27': '96f370cbd771a847ff9d37d000037d18dda7d113649036a403842c09e610606ac04217ad651c66edb3e66a8d65e66ad2',
                        '28': 'acc9e9dc5d1498c1a5cf2fa977298bb04b16d1ffe38f9ed7ca076fd55bd5bb2a2dbdc21d03640763a90ca66ced3a29aa',
                        '29': '8be35cd7d9d7081147eace4ea3a41609f65fcad2677dbe659a2b2f7673cb9b986531817c314b0d9d1f93a5ee7c176b1e',
                        '3': '995b0f1a6949b13d52e31737b09bc9af73e8d86824c742c94bb0f35e39f8d21ffcc0dc55d3d7351a245986e0e814124c',
                        '30': '838e6999e280483328b0893b74ac34268ed6d4c1c291cd6212f155cf959ffe96f4ed1c7dbec093125c5ff60cad253ab3',
                        '31': '81598ae491efa7dc792da276f90c055dbfe54684e0718ed34f829bbf62966fdc177464c3611c81c5c912261d2f70814f',
                        '32': '9962157960a16461b89c06aaf91d0e0817a650dfff74f041fe9100288769489e6cefe0aa501f2f165a53e5e5ba986f18',
                        '33': '93595a99565efdf519dfacfb0682654a2652a53756db3ecf213cf28faaafd96bdf48450667690c48b9cc9a7ff08314e0',
                        '34': '81cf17ed4079b9c601accec23f87414cee32baf86edfb8050fb8d980d0cd346767190f8735cde6be587c0cae6de25f29',
                        '35': 'aba9f2f9f74bf3cf053a18c2b4f1ac5a3cdf23ae9e45901a30385b765e1ed2ecd1e41edf89de5bc9fee777fec3b49971',
                        '36': '8ed3dd2dc68b43a35ce6900eeda5382ce9dd71fd821a6375739ee07de8d3674d8e263c760986396822f4d9f14043c3fe',
                        '37': 'b68dd494a35bb87cb047d87e72ad575de648a6cf7b556774e5a6d555222f3233d8e659970a49ea4f47494bc2aa317442',
                        '38': 'aa46d7ad828715f4b0cf1569d2e2e6c2295bf6cd9d0be41c12216f94fa3b15892f31378e71d83ca5c4d0bdf5d499d00a',
                        '39': 'a6de4853510dcb06a0a090f60144367a52ef3a03c3ee8e85c7d44a57e0126fe039d573e5362f1cdaa321f864411e75a4',
                        '4': '0c7a4899caf6410f97611949d094e652aa9622aa661901192bfc45c3261cb629f5f6b8ad3eb1143cd35713aef755f24a',
                        '5': '95844a49530a5c89ea4f1eb9f1e719ec156b209495081787d0aa2d9cce95440980d511ee3d8c6101d6fb77651db0375b',
                        '6': '050e37e6856a20b07a39fe254578afb1b668dcf5984e1a1a7120d8c0f2c2ecf8aad217045e1da626e95a96a029a6bdc3',
                        '7': '11f6c0c602badeaa267c7ed7efbe1dff45e9f9e3639f8c4049508745fce2dac66c2fbc193e0212836157a58014133291',
                        '8': '95040dc97d090070934c90d294ecdaf7fedb19e4c790777750671838613457e53cfc948133e0df75c5121ea1680c4c53',
                        '9': '0bfde7c4270f18ff0e90136eefbaf631a7a922f4b183295fbbb233dfc2071a2e6ec1fae1cbda1a86d2a86b86fd12567a'}}

```

**Comment by ChihChengLiang** [comment]
Should be fixed in #791

**Comment by ralexstokes** [comment]
closing this issue as I think we have sorted this out -- feel free to re-open if i'm missing something
